hcloud/client.go

```diff
@@ -169,15 +178,26 @@ func (c *Client) Do(r *http.Request, v interface{}) (*Response, error) {
 			return nil, err
 		}
 		response := &Response{Response: resp}
-
 		body, err := ioutil.ReadAll(resp.Body)
 		if err != nil {
 			resp.Body.Close()
 			return response, err
 		}
 		resp.Body.Close()
 		resp.Body = ioutil.NopCloser(bytes.NewReader(body))
-
+		if c.debugFile != nil {
+			fmt.Fprintln(c.debugFile, "------- HCLOUD_GO_DEBUG -------")
+			// Print Request Details
+			fmt.Fprintf(c.debugFile, "Request: %s - %s\n", r.Method, r.URL)
+			fmt.Fprintf(c.debugFile, "Request Headers: %#v\n", r.Header)
+			fmt.Fprintf(c.debugFile, "Request Body: %s\n", r.Body)
+			fmt.Fprintln(c.debugFile)
+			// Print Response Details
+			fmt.Fprintf(c.debugFile, "Response Status Code: %v\n", resp.StatusCode)
+			fmt.Fprintf(c.debugFile, "Response Headers: %#v\n", resp.Header)
+			fmt.Fprintf(c.debugFile, "Response Body: %s\n", body)
+			fmt.Fprintf(c.debugFile, "-------------------------------")
+		}
```

Comment:
What about using [`httputil.DumpRequest`](https://golang.org/src/net/http/httputil/dump.go?s=5638:5700#L181) and [`httputil.DumpResponse`](https://golang.org/src/net/http/httputil/dump.go?s=8166:8231#L271)?